QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,145 @@
+This document is a declaration of software quality for the `foonathan_memory` package imported by `foonathan_memory_vendor`, based on the guidelines in [REP-2004](https://www.ros.org/reps/rep-2004.html).
+
+# `foonathan_memory` Quality Declaration
+
+This quality declaration claims that the third party package `foonathan_memory` is in the **Quality Level 4** category.
+
+Below are the rationales, notes, and caveats for this claim, organized by each requirement listed in the [Package Requirements for Quality Level 4 in REP-2004](https://www.ros.org/reps/rep-2004.html).
+
+## Version Policy [1]
+
+### Version Scheme [1.i]
+
+`foonathan_memory` does not have a declared versioning scheme.
+
+### Version Stability [1.ii]
+
+`foonathan_memory` is in the beta stage, and currently has a version less than 1.0. The public API may change between minor version updates.
+
+### Public API Declaration [1.iii]
+
+All symbols in the installed headers are considered part of the public API with the exception of the `detail` directory and corresponding namespace.
+
+All installed headers are in the `include` directory of the package, headers in any other folders are not installed and considered private.
+
+### API Stability Within a Released ROS Distribution [1.iv]/[1.vi]
+
+`foonathan_memory` is a third party package and does not follow the ROS distribution timeline.
+It may introduce breaking API changes within a ROS distribution.
+To avoid breaking API within a ROS distribution, its vendor package `foonathan_memory_vendor` shall import a specific version.
+
+### ABI Stability Within a Released ROS Distribution [1.v]/[1.vi]
+
+`foonathan_memory` is a third party package and does not follow the ROS distribution timeline.
+It may introduce breaking ABI changes within a ROS distribution.
+To avoid breaking API within a ROS distribution, its vendor package `foonathan_memory_vendor` shall import a specific version.
+
+## Change Control Process [2]
+
+`foonathan_memory` does not have a stated change control process.
+
+### Change Requests [2.i]
+
+This package does not require that changes occur through a pull request.
+Many changes are committed directly by its maintainer.
+
+### Contributor Origin [2.ii]
+This package does not have a confirmation of contributor origin policy.
+
+### Peer Review Policy [2.iii]
+
+There is no stated peer review policy, but each pull request is at least reviewed by the core maintainer.
+
+### Continuous Integration [2.iv]
+
+Changes must pass CI on AMD 64 for:
+- Ubuntu 16.04 for gcc 4.9 and 5.5
```

Comment:
Ubuntu 16.04 ?  we are targeting Focal, (20.04) right?

Reply:
My understanding is that this is documenting foonathan_memory's CI upstream. Which is using 16.04.

Reply:
Err, I see why its confusing since I just named this QUALITY_DECLARATION.md. I renamed the file to reflect that it's for foonathan_memory itself, and not the _vendor package.